### PR TITLE
enhance envbinding: support apply resources to cluster 

### PR DESCRIPTION
**Pull request by yangsoon**
**What this PR does / why we need it**:

this PR enhance the ability of envbinding.

1. add SingleClusterEngine which support apply patched Application to local cluster
2. add `ouputResourcesTo` field, user can specify the name and ns of the configMap where the resource is stored.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml [hunk 1]
```diff
@@ -0,0 +1,341 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (wonderflow) [inline pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml:1]: 这个文件用到了吗

File: pkg/controller/standard.oam.dev/v1alpha1/envbinding/engine.go [hunk 2]
```diff
@@ -89,11 +93,28 @@ func (o *OCMEngine) Prepare(ctx context.Context, configs []v1alpha1.EnvConfig) e
 	return nil
 }
 
+func (o *OCMEngine) initEnvBindApps(ctx context.Context, envBinding *v1alpha1.EnvBinding, baseApp *v1beta1.Application, appParser *appfile.Parser) ([]*EnvBindApp, error) {
+	envBindApps, err := CreateEnvBindApps(envBinding, baseApp)
+	if err != nil {
+		return envBindApps, err
```
Comment (wonderflow) [inline pkg/controller/standard.oam.dev/v1alpha1/envbinding/engine.go:99]: 如果有err就 return nil,err 。 除非这个 envBindApps 在err反馈的情况下还有用，代码不要引起歧义

File: pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml (hunk 1 above)
Comment (yangsoon) [reply pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml:1]: This file is used in the test

File: pkg/controller/standard.oam.dev/v1alpha1/envbinding/engine.go (hunk 2 above)
Comment (yangsoon) [reply pkg/controller/standard.oam.dev/v1alpha1/envbinding/engine.go:99]: got it

File: pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml (hunk 1 above)
Comment (wonderflow) [reply pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml:1]: 没看到test里面有用到

Comment (yangsoon) [reply pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml:1]: https://github.com/oam-dev/kubevela/blob/ad9f196f59c4501e32f12cd88acfc47a9cb8e514/pkg/controller/standard.oam.dev/v1alpha1/envbinding/suit_test.go#L59

https://github.com/oam-dev/kubevela/blob/ad9f196f59c4501e32f12cd88acfc47a9cb8e514/pkg/controller/standard.oam.dev/v1alpha1/envbinding/envbinding_controller_test.go#L420-L425

Comment (wonderflow) [reply pkg/controller/standard.oam.dev/v1alpha1/envbinding/testdata/crds/manifestwork.yaml:1]: ok

**Pull request closed without merging**